### PR TITLE
Slightly improve handling of keys

### LINKED ISSUE #43: cm does not handle Shift+TAB to scroll backwards through the panels
introduced in #42

### PR DESCRIPTION
**Pull request by et342**
~~Depends on #53
(It doesn't actually depends on it, it just conflicts, for some reason, if done separately, and I couldn't find a way how to solve this, I'm sorry)~~

Closes #43

### CODE REVIEW

**Comment by rexim** [comment]
@et342 hey thank you so much for trying to improve the code I really appreciate it! 

Could you please provide the criteria by which you'd claim this an improvement? So far the code does the same thing, but it became more complex. My brain is too small to understand complex constructions (I really apologize for that, I can't help it!). So I need to know if such complexity increase is actually justified. Thanks!

**Comment by et342** [comment]
@rexim 

So `getch()` may return codes that exceed 0xFF and don't correspond to literal characters. This means that simply truncating those codes to `u8` isn't correct, and while works for now, won't work in the future (when there will be a need to handle keys other than letters).

Idea here is to wrap `getch` codes into the `Key` enum to separate single-byte codes from multi-byte, so it is possible to use literal characters, in code, when expecting character keys, and possible to use ncurses KEY_* constants and other arbitrary codes ([ncurses man](https://invisible-island.net/ncurses/man/define_key.3x.html) says you can define you own keycodes) in a single match expression.

So basically, improvement here is that non-character key codes are not truncated and can be handled, and the rest of it is basically for syntax because Rust won't let you match ints' using characters, so you'll have to sprinkle some `if`s and `as`es and whatnot around, instead of just

```rust
match key {
    'A' => ...
    KEY_CODE => ...
    'B' | KEY_C => ...
}
```

**Comment by et342** [comment]
I've decided to close this as I realized that, since keys are supposed to be customizable, there will a different approach to handling them, and code here will have no use.
